src/com/owncloud/android/authentication/AuthenticatorAsyncTask.java

```diff
@@ -72,18 +72,32 @@ protected RemoteOperationResult doInBackground(Object... params) {
             );
             result = operation.execute(client);
 
+            String permanentLocation = null;
             if (operation.wasRedirected()) {
                 RedirectionPath redirectionPath = operation.getRedirectionPath();
-                String permanentLocation = redirectionPath.getLastPermanentLocation();
-                result.setLastPermanentLocation(permanentLocation);
+                permanentLocation = redirectionPath.getLastPermanentLocation();
             }
 
             // Operation - get display name
             if (result.isSuccess()) {
-                GetRemoteUserInfoOperation remoteUserNameOperation = new GetRemoteUserInfoOperation();
-                result = remoteUserNameOperation.execute(client);
+                if (permanentLocation == null) {
+                    GetRemoteUserInfoOperation remoteUserNameOperation = new GetRemoteUserInfoOperation();
+                    result = remoteUserNameOperation.execute(client);
+                } else {
+                    // we can't assume that any subpath of the domain is correctly redirected; ugly stuff
+                    GetRemoteUserInfoOperation remoteUserNameOperation = new GetRemoteUserInfoOperation();
+                    OwnCloudClient redirectedClient = OwnCloudClientFactory.createOwnCloudClient(
+                        Uri.parse(AccountUtils.trimWebdavSuffix(permanentLocation)),
+                        mContext,
+                        true
```

Comment:
With the cursor over the name of the method, click F1. The documentation of the method will be shown, including the name of the parameter and its meaning.

Declaring an assigning a variable to use it just once as a parameter seems too much IMHO.